py-polars/src/conversion/any_value.rs

```diff
@@ -115,7 +115,7 @@ pub(crate) fn any_value_into_py_object(av: AnyValue, py: Python) -> PyObject {
 }
 
 type TypeObjectPtr = usize;
-type InitFn<'py> = fn(&Bound<'py, PyAny>, bool) -> PyResult<AnyValue<'py>>;
```

Comment:
This made the lifetime fixed to `'py`. But given that we put `InitFn` in a global LUT, the materialization of `'py` is not bounded to the python lifetime, but `'static`, as the LUT lives for the entirety of the program.

That's why we need to use `for<'py>` to inform the compiler that the lifetime will be initiated on function call and is not bound to the function itself.